vue/wolox-books/src/router.js

```diff
@@ -1,20 +1,37 @@
 import Vue from 'vue'
 import Router from 'vue-router'
+import routes from './routes'
+import { getToken } from './services/LocalStorageService'
 
 Vue.use(Router)
 
-export default new Router({
+const router = new Router({
   routes: [
-    { path: '/', redirect: '/login' },
     {
-      path: '/sign_up',
+      path: routes.home,
+      name: 'home',
+      component: () => import(/* webpackChunkName: "home" */ './views/Home.vue')
+    },
+    {
+      path: routes.sign_up,
       name: 'sign_up',
       component: () => import(/* webpackChunkName: "sign_up" */ './views/Register.vue')
     },
     {
-      path: '/login',
+      path: routes.login,
       name: 'login',
       component: () => import(/* webpackChunkName: "login" */ './views/Login.vue')
+    },
+    {
+      path: '*',
+      redirect: routes.home
     }
   ]
 })
+
+router.beforeEach((to, from, next) => {
+  const token = getToken()
+  to.fullPath === routes.login || to.fullPath === routes.sign_up ? next() : !token ? next(routes.login) : next()
```

Comment:
missing return here?

Reply:
![image](https://user-images.githubusercontent.com/26824183/56243669-c6cd9700-6060-11e9-9b6a-197177a33fd6.png)
for it I use next()